runtime/gc_vlhgc/ConfigurationIncrementalGenerational.cpp

```diff
@@ -276,11 +276,10 @@ MM_ConfigurationIncrementalGenerational::initialize(MM_EnvironmentBase *env)
 	env->disableHotFieldDepthCopy();
 
 	if (result) {
-		if (MM_GCExtensions::OMR_GC_SCAVENGER_SCANORDERING_DYNAMIC_BREADTH_FIRST != extensions->scavengerScanOrdering) {
-			extensions->scavengerScanOrdering = MM_GCExtensions::OMR_GC_SCAVENGER_SCANORDERING_BREADTH_FIRST;
-		} else {
-			extensions->adaptiveGcCountBetweenHotFieldSort = false;
+		if (MM_GCExtensions::OMR_GC_SCAVENGER_SCANORDERING_NONE == extensions->scavengerScanOrdering) {
+			extensions->scavengerScanOrdering = MM_GCExtensions::OMR_GC_SCAVENGER_SCANORDERING_DYNAMIC_BREADTH_FIRST;
 		}
+		extensions->adaptiveGcCountBetweenHotFieldSort = false;
```

Comment:
should this flag be set inside the 'if'?
what is somebody explicitly sets DBF and sorting to true? seems like sorting DBF will be obeyed but sorting not?

Reply:
No it should be outside the if statement, as currently it is enabled on extensions initialization, however, it was only beneficial for gencon with DBF scan ordering, so should be disabled in balanced. For balanced, regardless of the scan ordering selection, this flag should not be set. This also handles the case where the user adds the option "dynamicBreadthFirstScanOrdering". This would result in the IF statement being false, however, the flag still needs to be disabled. Overall design could be changed to disable this flag by default in extensions initialization and only enable it if DBF scan ordering is set on gencon.